pkg/asset/machines/openstack/machinesets.go

```diff
@@ -47,7 +47,19 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 		if int32(idx) < total%numOfAZs {
 			replicas++
 		}
-		provider, err := generateProvider(clusterID, platform, mpool, osImage, az, role, userDataSecret, trunkSupport, volumeAZs[idx%len(volumeAZs)])
+		provider, err := generateProvider(
+			clusterID,
+			platform,
+			mpool,
+			osImage,
+			role,
+			userDataSecret,
+			trunkSupport,
+			openstack.FailureDomain{
+				ComputeAvailabilityZone: az,
+				StorageAvailabilityZone: volumeAZs[idx%len(volumeAZs)],
```

Comment:
We pass exactly the same data as before (no failure domains for Compute nodes!). We just pack them in an `openstack.FailureDomain` for consistency with masters. I think it's better than passing regular strings.